packages/ember-views/lib/views/view.js

```diff
@@ -1542,7 +1543,13 @@ View.views = {};
 // method.
 View.childViewsProperty = childViewsProperty;
 
+const DeprecatedView = View.extend({
+  init() {
+    this._super(...arguments);
+    Ember.deprecate(`Ember.View is deprecated. Consult the Deprecations Guide for a migration strategy.`, !!Ember.ENV.ENABLE_LEGACY_VIEW_SUPPORT, { url: 'http://emberjs.com/deprecations/v1.x/#toc_ember-view' });
```

Comment:
This is likely annoying, but I would prefer that the flag itself is also private.  Can you underscore?